### PR TITLE
Fda metrics split on sequence

### PR DESCRIPTION
**Pull request by johnegarza**
This PR introduces a CWL scatter across individual sequence elements (a bam or pair of fastqs) of the input array for each of the 3 unaligned inputs (normal DNA, tumor DNA, tumor RNA) when calculating statistics to be reported to the FDA in clinical cases. Previously, some statistics were calculated using data from all sequence files for an input at once, which is less accurate.

### CODE REVIEW

**Comment by gschang** [comment]
Thank you for this work, John. These updates are about FDA QC metric reports, which John and we recently added to the immuno workflow. We discussed more consistent QC reporting with multiple instrument data, i.e. towards more generalized interface to include them in the report.

As John suggests, I plan to run a test to doublecheck the key changes in the final FDA QC metrics report.

File: definitions/tools/unaligned_seq_fda_stats.cwl [hunk 1]
```diff
@@ -121,6 +117,22 @@ requirements:
                 sub Main {
                     # local variables
                     
+                    
+                    # gets the input file format
+                    unless (defined $format)
```
Comment (tmooney) [inline definitions/tools/unaligned_seq_fda_stats.cwl:122]: It looks like this is being defined on line 72.  Will we ever fall into this case?  If we don't fall into this case for whatever reason, does that mean the user must NOT pass the format lest it be added to `@paths` instead?

File: definitions/tools/unaligned_seq_fda_stats.cwl [hunk 2]
```diff
@@ -251,6 +262,15 @@ requirements:
                     }
                     
                     
+                    # prints out the source file information
+                    printf "\n\n[Input file information]";
+                    foreach my $path (@paths)
+                    {
+                        my ($name, $dir, $ext) = fileparse($path, qr/\.[^.]*/);
+                        printf "\n%s\t%s", $name . $ext, $dir;
```
Comment (tmooney) [inline definitions/tools/unaligned_seq_fda_stats.cwl:270]: If you don't care about the extension, can simplify this:
```suggestion
                        my ($name, $dir) = fileparse($path);
                        printf "\n%s\t%s", $name, $dir;
```

**Review by tmooney — COMMENTED** [review]
In general I'm a little puzzled by some of the commented-out code and related conditionals in these scripts.  Seems like it makes the logic more complicated and invites the potential for incorrect behavior if the wrong combination of lines are uncommented.

File: definitions/tools/unaligned_seq_fda_stats.cwl (hunk 2 above)
Comment (gschang) [reply definitions/tools/unaligned_seq_fda_stats.cwl:270]: These comments are good points. I am going to clean them up in the source codes as Tom suggests, and also removing comment lines that are not in the right place.

**Comment by gschang** [comment]
I just want to update this ticket here and John, Jasreet, and I are working on this PR. John and we have discussed more upgrades for generalized workflow concept, and now in progress. I updates the scripts as Tom advised above and finished testing.

**Comment by gschang** [comment]
These new changes in the `generate_fda_metrics.cwl` subworkflow worked on my end. I doublechecked every upgrade/fixation.

We're going to release a new immuno workflow unless there are other comments from someone else. I have requested John to create a new immuno processing profile in two versions--(1) immuno.cwl (currently running) and (2) immuno.cwl without rnaseq input.

**Review by gschang — APPROVED** [review]

**Pull request merged**